libs/libogg-1.3.2/include/ogg/os_types.h

```diff
@@ -75,7 +75,7 @@
    typedef uint32_t ogg_uint32_t;
    typedef int64_t ogg_int64_t;
 
-#elif defined(__HAIKU__)
+#elif defined(__HAIKU__) || defined(__GNUC__) || defined(__clang__)
```

Comment:
I think I understand the reasoning behind this change, and usually I prefer these platform checks for type definitions here in the library's code instead of in build tool configuration, in my own stuff. But I don't really like the idea of modifying a third party library's code/headers if we don't have to, since it makes it harder to upgrade in the future etc.

Would it be possible to make the cmake file generate the config_types.h file that this library seems to expect on Linux, instead of modifying the header code?